### ISSUE TITLE
Data Holdings tables omit some models

### ISSUE THREAD

**Issue by taylor13** [PCMDI/pcmdi.github.io#268]
@mauzey1  @sashakames It looks like the data availability tables (https://pcmdi.llnl.gov/CMIP6/ArchiveStatistics/esgf_data_holdings/)  are listing only 8 models, but more than a dozen models have contributed output.  It was working a few days ago I think, and I don't think any changes have been made, so this is a bit of a mystery.

**Comment by taylor13** [comment]
@mauzey1 @sashakames   This has be corrected.  If you didn't have to do anything to correct it, is there a way of preventing a temporary issue of this kind in the future?

**Comment by mauzey1** [comment]
@taylor13  @sashakames 
I'm not sure what happened that made only 8 models show up.  The script that produces these tables can be a bit slow, which is mostly due to waiting for search queries to complete.

I do have a new script that will add the new variable count and sampling frequency tables to the page.  It also uses more efficient ways of querying ESGF so it might mitigate these kinds of issues.

https://github.com/ESGF/esgf-utils/pull/5

**Comment by taylor13** [comment]
@mauzey1 @sashakames  Sounds good.  Let's not spend too much time diagnosing a problem that may only rarely occur.  If we find it happens again, we can re-evaluate.  

That being said, is your search querying only the PCMDI node or does it go out to other nodes too?  If it looks elsewhere and a node is temporarily down, will it miss populating the table?  What if PCMDI's node is down; will it construct a new table based on what was last returned by a successful search, or will it produce an empty table?

**Comment by mauzey1** [comment]
@taylor13 @sashakames 
The script will query Solr on esgf-node.llnl.gov and all of its shards similar to how esg-search does queries.  

Currently, the script is designed to overwrite an HTML file every time it runs and creates new tables.  If the script were to run and encounter an error when trying to query esgf-node.llnl.gov, it would create a blank page.  Future versions of the script could be made to write to a file only after successfully querying esgf-node.llnl.gov.

**Comment by taylor13** [comment]
guess we should consider doing that, but perhaps we should wait do see how often the problem arises.  Perhaps @sashakames has some guidance.

**Comment by durack1** [comment]
@mauzey1 the way I have done this in the past (with indexing CMIP5 data) was to set a minimum value, and if this is not met (within tolerances) then I quit and try again later. With CMIP6 I really only see this number increasing for the next couple of years, so we could implement such a test.

This could also be rolled into the "database" I was suggesting which provides a time evolving perspective of the growth of the CMIP6 archive, it'd be useful to know about this in coming years

**Comment by durack1** [comment]
@mauzey1 rather than opening a new issue, I am reusing this (which I think can be closed once we resolve my issue, as I believe the standing issue noted by the title is solved).

It seems the page https://pcmdi.llnl.gov/CMIP6/ArchiveStatistics/esgf_data_holdings/ was last updated on the 28th March 2020. Does that mean that there are no new updates since that time, or something has gone wrong with the updated page generation?

**Comment by mauzey1** [comment]
@durack1 I am not sure what happened on March 28 that would make the page not update.  I am currently getting help from Tony to get access to the server (I cannot SSH into it for some reason).

**Comment by durack1** [comment]
Thanks @mauzey1, I’ll be interested to hear and update when you have one

**Comment by durack1** [comment]
@mauzey1 @ahoang1 just wondering if we have any update on this, I have started receiving queries from folks who have also noted the updates have stalled

**Comment by mauzey1** [comment]
@durack1 The [ESGF CMIP6 Data Holdings page](https://pcmdi.llnl.gov/CMIP6/ArchiveStatistics/esgf_data_holdings/) is now updating again.

**Comment by durack1** [comment]
@mauzey1 great thanks for this, I'll close the issue as I think we are now all sorted
